cms/static/js/views/course_outline.js

```diff
@@ -209,6 +212,33 @@ define(["jquery", "underscore", "js/views/xblock_outline", "js/views/utils/view_
                         ensureChildrenRendered: this.ensureChildrenRendered.bind(this)
                     });
                 }
+            },
+
+            handleReIndexEvent: function(event) {
+                var self = this;
+                event.preventDefault();
+                var target = $(event.currentTarget);
+                target.css('cursor', 'wait');
+                this.startReIndex()
+                    .done(function() {self.onIndexSuccess();})
+                    .always(function() {target.css('cursor', 'pointer');});
+            },
+
+            startReIndex: function() {
+                var locator =  window.course.id;
+                return $.ajax({
+                    url: '/course_index/' + locator,
+                    method: 'GET'
+                    });
+            },
+
+            onIndexSuccess: function() {
+                var msg = new AlertView.Announcement({
+                        title: gettext('Course Index'),
+                        message: gettext('Course has been successfully reindexed.'),
+                        intent: 'announcement'
```

Comment:
Nitpick: if you're specifying `AlertView.Announcement`, then you don't need to also specify `intent: 'annoucement'` here.